modules/java_api/samples/face_detection_java_api_2.0_sample/Main.java

```diff
@@ -0,0 +1,126 @@
+import org.intel.openvino.*;
+import org.intel.openvino.Core;
+import org.opencv.core.*;
+import org.opencv.highgui.HighGui;
+import org.opencv.imgcodecs.*;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/*
+This is face detection java sample.
+
+Upon the start-up the sample application reads command line parameters and loads a network
+and an image to the Inference Engine device. When inference is done, the application will show
+the image with detected objects enclosed in rectangles in new window.It also outputs the
+confidence value and the coordinates of the rectangle to the standard output stream.
+
+To get the list of command line parameters run the application with `--help` paramether.
+*/
+public class Main {
+    public static void main(String[] args) {
+        final double THRESHOLD = 0.7;
+        try {
+            System.loadLibrary(org.opencv.core.Core.NATIVE_LIBRARY_NAME);
+        } catch (UnsatisfiedLinkError e) {
+            System.err.println("Failed to load OpenCV library\n" + e);
+            System.exit(1);
+        }
+        try {
+            System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+        } catch (UnsatisfiedLinkError e) {
+            System.err.println("Failed to load OpenVINO library\n" + e);
+            System.exit(1);
+        }
+
+        ArgumentParser parser = new ArgumentParser("This is face detection sample");
+        parser.addArgument("-i", "path to image");
+        parser.addArgument("-m", "path to model .xml");
+        parser.parseArgs(args);
+
+        String imgPath = parser.get("-i", null);
+        String xmlPath = parser.get("-m", null);
+
+        if (imgPath == null) {
+            System.out.println("Error: Missed argument: -i");
+            return;
+        }
+        if (xmlPath == null) {
+            System.out.println("Error: Missed argument: -m");
+            return;
+        }
+
+        Mat image = Imgcodecs.imread(imgPath);
+
+        Core core = new Core();
+        Model net = core.read_model(xmlPath);
+
+        int[] dimsArr = { 1, image.rows(), image.cols(), 3 };
+        Tensor input_tensor = new Tensor(ElementType.u8, dimsArr, image.dataAddr());
+
+        int[] tensor_shape = input_tensor.get_shape(); // Do we need Shape?
+        Layout tensor_layout = new Layout("NHWC");
+
+        PrePostProcessor ppp = new PrePostProcessor(net);
+
+        ppp.input()
+                .tensor()
+                .set_element_type(ElementType.u8)
+                .set_layout(tensor_layout)
+                .set_spatial_static_shape(tensor_shape[Layout.height_idx(tensor_layout)],
+                                       tensor_shape[Layout.width_idx(tensor_layout)]);
```

Comment:
Do we always need set `set_spatial_static_shape`?

Reply:
Perhaps there is another way to set the shape, but I didn't find it. Example taken from [hello_classification/main.cpp](https://github.com/openvinotoolkit/openvino/blob/master/samples/cpp/hello_classification/main.cpp)

Reply:
Maybe it make sense just use `image.rows()` and `image.cols()`